sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py

```diff
@@ -0,0 +1,378 @@
+                                                     client_kwargs=client_kwargs)
+        self.parameter_name = parameter_name
+
+    def create_resource(self, name, **kwargs):
+        if self.is_live:
+            self.client = self.create_mgmt_client(ServiceBusManagementClient)
+            group = self._get_resource_group(**kwargs)
+            namespace = self._get_namespace(**kwargs)
+            topic_async_operation = self.client.topics.create_or_update(
+                group.name,
+                namespace.name,
+                name,
+                {}
+            )
+            self.resource = topic_async_operation
+        else:
+            self.resource = FakeResource(name=name, id=name)
+        return {
+            self.parameter_name: self.resource,
+        }
+
+    def remove_resource(self, name, **kwargs):
+        if self.is_live:
+            group = self._get_resource_group(**kwargs)
+            namespace = self._get_namespace(**kwargs)
+            self.client.topics.delete(group.name, namespace.name, name, polling=False)
+
+
+
+class ServiceBusSubscriptionPreparer(_ServiceBusChildResourcePreparer):
+    def __init__(self,
+                 name_prefix='',
+                 parameter_name=SERVICEBUS_SUBSCRIPTION_PARAM,
+                 resource_group_parameter_name=RESOURCE_GROUP_PARAM,
+                 servicebus_namespace_parameter_name=SERVICEBUS_NAMESPACE_PARAM,
+                 servicebus_topic_parameter_name=SERVICEBUS_TOPIC_PARAM,
+                 disable_recording=True, playback_fake_resource=None,
+                 client_kwargs=None):
+        super(ServiceBusSubscriptionPreparer, self).__init__(name_prefix,
+                                                     resource_group_parameter_name=resource_group_parameter_name,
+                                                     servicebus_namespace_parameter_name=servicebus_namespace_parameter_name,
+                                                     disable_recording=disable_recording,
+                                                     playback_fake_resource=playback_fake_resource,
+                                                     client_kwargs=client_kwargs)
+        self.servicebus_topic_parameter_name = servicebus_topic_parameter_name
+        self.parameter_name = parameter_name
+
+    def create_resource(self, name, **kwargs):
+        if self.is_live:
+            self.client = self.create_mgmt_client(ServiceBusManagementClient)
+            group = self._get_resource_group(**kwargs)
+            namespace = self._get_namespace(**kwargs)
+            topic = self._get_topic(**kwargs)
+            subscription_async_operation = self.client.subscriptions.create_or_update(
+                group.name,
+                namespace.name,
+                topic.name,
+                name,
+                {}
+            )
+            self.resource = subscription_async_operation
```

Comment:
Is this an async operation? If so we need to use `subscription_async_operation.result()`, and if it's not we could probably just assign self.resource directly.

Reply:
Same question for the other child resources

Reply:
Good catch.  Was spurious.